clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -94,6 +95,85 @@ def SInt16 : SInt<16>;
 def SInt32 : SInt<32>;
 def SInt64 : SInt<64>;
 
+//===----------------------------------------------------------------------===//
+// FloatType
+//===----------------------------------------------------------------------===//
+
+def CIR_FloatType : CIR_Type<"Float", "float",
+    [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>]> {
+  let summary = "CIR floating-point type";
+  let description = [{
+    CIR type that represents C/C++ floating-point types, including `float`, \
+    `double`, and `long double`.
+  }];
+  let parameters = (ins "::mlir::cir::FloatType::Format":$format);
+
+  let builders = [
+    TypeBuilder<(ins CArg<"::mlir::cir::FloatType::Format">:$format)>
+  ];
+
+  let hasCustomAssemblyFormat = 1;
+  let genVerifyDecl = 1;
+
+  let extraClassDeclaration = [{
+    // Floating-point format.
+    enum Format : uint32_t {
+      FF_Binary16,   // IEEE-754 binary16 format
+      FF_Binary32,   // IEEE-754 binary32 format
+      FF_Binary64,   // IEEE-754 binary64 format
+      FF_Binary64E,  // IEEE-754 binary64-extended format
```

Comment:
I would much prefer a different name for this, one that makes clear this is the x86 80-bit format.  The `Binary64E` name is not precise enough, since C compilers on non-x86 systems could specify that the binary64-extended format used for `_Float64x` is IEEE 128-bit.  LLVM IR uses `x86_fp80` for the name of this type.  I think something along those lines would be appropriate here.

Reply:
Yes, I feel the name is a bit confusing too. I'll change `FF_Binary64E` to `FF_X86Fp80`.

Reply:
Renamed `FF_Binary64E` to `FF_X86Fp80`.